### ISSUE TITLE
table-alternatives showing too many rows from association tables

### ISSUE THREAD

**Issue by ljpearlman** [informatics-isi-edu/ermrestjs#366]
If I have two tables, `X` and `Y` linked by an association table `ASSOC` (i.e., with foreign keys from `ASSOC` to both `X` and `Y`), and I declare the table `ALT_X` (which has a foreign key to `X`) as an alternate view of `X`, then viewing any row of `Y` in the `record` app will display all rows of `ALT_X`, not just the relevant ones.

That's pretty hard to parse, so here's a concrete example. If I define these tables:

```
create schema test;
set search_path = test;

create table x (id integer primary key, xval text);
create table y (id integer primary key, yval text);
create table assoc (xid integer references x(id), yid integer references y(id), primary key (xid, yid));
create table alt_x(id integer primary key references x(id), alt_xval text);

insert into x (id, xval) values (1, 'one'), (2, 'two'), (3, 'three');
insert into alt_x (id, alt_xval) select id, 'alt ' || xval from x;
insert into y (id, yval) values (10, 'one y'), (11, 'another one y'), (20, 'two y');
insert into assoc(xid, yid) values (1, 10), (1, 11), (2, 20);

insert into _ermrest.model_table_annotation(schema_name, table_name, annotation_uri, annotation_value)
   values ('test', 'x', 'tag:isrd.isi.edu,2016:table-alternatives',
            '{"compact/brief" : ["test", "alt_x"]                                                                                   
             }');
```

Then, if I viewed record 10 of `y` in the `record` app (`.../record/#N/test:y/id=10`), I would expect to see one row from `alt_x` (with `id`=1 and `alt_xval`="alt one"). Instead, I see all three rows from `alt_x`.

**Comment by hongsudt** [comment]
@RFSH can this be closed?

**Comment by RFSH** [comment]
@hongsudt We haven't addressed this issue yet.

**Comment by ljpearlman** [comment]
The version Aref tested on rbk-dev yesterday looked good to me (not closing the bug because it doesn't look like it's been pushed to master yet).

**Comment by RFSH** [comment]
@ljpearlman I pushed it to master now. It should be fixed.
